velox/type/Timestamp.h

```diff
@@ -49,22 +70,42 @@ struct Timestamp {
   }
 
   int64_t toNanos() const {
-    // int64 can store around 292 years in nanos ~ till 2262-04-12
-    // The addition cannot overflow because the product will be promoted to
-    // uint64_t first and its value is at most UINT64_MAX / 2.
-    return checkedMultiply(seconds_, (int64_t)1'000'000'000) + nanos_;
+    // int64 can store around 292 years in nanos ~ till 2262-04-12.
+    // When an integer overflow occurs in the calculation,
+    // an exception will be thrown.
+    try {
+      return checkedPlus(
+          checkedMultiply(seconds_, (int64_t)1'000'000'000), (int64_t)nanos_);
+    } catch (...) {
```

Comment:
perhaps, const std::exception& e

then, use e.what() instead of hard-coding "integer overflow"

This way we ensure we get correct message even if exception is different from overflow.